### PR TITLE
Updated method for empty data check

### PR DESCRIPTION
**Pull request by tbcarver**
The method of checking for an empty array and setting an object is before the call to json encode.

### CODE REVIEW

**Review by bridger26 — APPROVED** [review]

File: src/SendEmail.php [hunk 1]
```diff
@@ -24,6 +24,10 @@ private function postRequest(): Response
     {
         $client = new Client();
 
+        if (!($this->data['data'] && count($this->data['data']))) {
+            $this->data['data'] = (object)[];
+        }
+
```
Comment (159TannerRobertson) [inline src/SendEmail.php:30]: Isn't this going to fail on the validate() function? if data is empty

**Review by 159TannerRobertson — APPROVED** [review]

File: src/SendEmail.php (hunk 1 above)
Comment (tbcarver) [reply src/SendEmail.php:30]: its a little confusing, but validate checks this->data, this is checking this->data->data set on buildEmailContent. I also tested with web app both empty and with an array and both went to sales force.

**Comment by tbcarver** [comment]
This feature of passing null to data.data isn't needed at this time.
